File: app/src/main/java/com/desuzed/clocknweather/util/ArrowImageView.java
package com.desuzed.clocknweather.util;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

import androidx.annotation.Nullable;

@SuppressLint("AppCompatCustomView")
public class ArrowImageView extends ImageView {
    private OnRotationChangedListener listener;

    public void setListener(OnRotationChangedListener listener) {
        this.listener = listener;
    }

    public ArrowImageView(Context context) {
        super(context);
    }

    public ArrowImageView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public ArrowImageView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setRotation(float rotation, int arrowIndex) {
        float oldRotation = this.getRotation();
        if (oldRotation != rotation) {
            listener.rotationChanged(rotation, arrowIndex);
        }
        super.setRotation(rotation);
    }
//метод для того, чтобы музыка не включалась при запуске приложения
    public boolean rotateArrow(float rotation) {
        boolean isChanged = false;
        float oldRotation = this.getRotation();
        if (oldRotation != rotation) {
            isChanged = true;
        }
        super.setRotation(rotation);
        return isChanged;
    }

    public interface OnRotationChangedListener {
        void rotationChanged(float rotation, int arrowIndex);
    }
}

File: app/src/main/java/com/desuzed/clocknweather/util/ArrowImageView.kt
package com.desuzed.clocknweather.util

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView

@SuppressLint("AppCompatCustomView")
class ArrowImageView : ImageView {
    private var listener: OnRotationChangedListener? = null
    fun setListener(listener: OnRotationChangedListener?) {
        this.listener = listener
    }

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
    }

    fun setRotation(rotation: Float, arrowIndex: Int) {
        val oldRotation = rotation
        if (oldRotation != rotation) {
            listener!!.rotationChanged(rotation, arrowIndex)
        }
        super.setRotation(rotation)
    }

    //метод для того, чтобы музыка не включалась при запуске приложения
    fun rotateArrow(rotation: Float): Boolean {
        var isChanged = false
        val oldRotation = this.rotation
        if (oldRotation != rotation) {
            isChanged = true
        }
        super.setRotation(rotation)
        return isChanged
    }

    interface OnRotationChangedListener {
        fun rotationChanged(rotation: Float, arrowIndex: Int)
    }
}